crates/ruff/src/rules/tryceratops/rules/raise_within_try.rs

```diff
@@ -77,15 +82,55 @@ pub(crate) fn raise_within_try(checker: &mut Checker, body: &[Stmt], handlers: &
         return;
     }
 
+    // The names of exceptions handled by the `try` Stmt. We can't compare `Expr`'s since they have
+    // different ranges, but by virtue of this function's call path we know that the `raise`
+    // statements will always be within the surrounding `try`.
+    let handler_ids: FxHashSet<&str> = helpers::extract_handled_exceptions(handlers)
+        .into_iter()
+        .filter_map(|handler| {
+            if let Expr::Name(ast::ExprName { id, .. }) = handler {
+                Some(id.as_str())
+            } else {
+                None
+            }
+        })
+        .collect();
+
     let raises = {
         let mut visitor = RaiseStatementVisitor::default();
         visitor.visit_body(body);
         visitor.raises
     };
 
     for stmt in raises {
-        checker
-            .diagnostics
-            .push(Diagnostic::new(RaiseWithinTry, stmt.range()));
+        let Stmt::Raise(ast::StmtRaise { exc, .. }) = stmt else {
+            continue;
+        };
+
+        let Some(exception) = exc else {
+            continue;
+        };
```

Comment:
Nit: Does the following work?
```suggestion
        let Stmt::Raise(ast::StmtRaise { exc: Some(exception), .. }) = stmt else {
            continue;
        };
```